qiime2/sdk/plugin_manager.py

```diff
@@ -79,20 +79,43 @@ def _integrate_plugin(self, plugin):
             self.transformers[input][output] = transformer_record
 
         for name, record in plugin.formats.items():
+
             if name in self.formats:
                 raise NameError(
                     "Duplicate format registration (%r) defined in plugins: %r"
                     " and %r" %
                     (name, record.plugin.name, self.formats[name].plugin.name)
                 )
             self.formats[name] = record
+
         self.type_formats.extend(plugin.type_formats)
 
     # TODO: Should plugin loading be transactional? i.e. if there's
     # something wrong, the entire plugin fails to load any piece, like a
     # databases rollback/commit
 
     @property
+    def importable_formats(self):
+        """Return dict of formats that are importable.
+
+        A format is importable if an installed plugin can transform it into
+        one of the canonical semantic types.
+
+        """
+        _importable_formats = {}
+        for plugin in self.plugins.values():
+            for name, record in plugin.formats.items():
+                for type_format in self.type_formats:
+                    from_type = transform.ModelType.from_view_type(
+                        record.format)
+                    to_type = transform.ModelType.from_view_type(
+                        type_format.format)
+                    if from_type.has_transformation(to_type):
+                        _importable_formats[name] = record.format
```

Comment:
Actually, I'm not going to do this for now because I think the benefit I get from leaving it in in order to work in `importable_types` probably outweighs the cost.  Once I push I'd like to know if you have any ideas RE how to achieve a similar affect using `while` and `pop()` or some other such mechanism.  Thanks!